core/domain/learner_progress_services.py

```diff
@@ -445,6 +447,61 @@ def mark_collection_as_incomplete(user_id, collection_id):
         _save_incomplete_activities(incomplete_activities)
 
 
+def add_topic_to_learn(user_id, topic_id):
+    """This function checks if the topic exists in the learnt.
+    If it does not exist we call the function in learner
+    goals services to add the topic to the learn list.
+
+    Args:
+        user_id: str. The id of the user.
+        topic_id: str. The id of the topic to be added to the
+            learner goals.
+
+    Returns:
+        (bool, bool). The first boolean indicates whether the topic
+        already exists in either of the "learnt topics" lists and
+        the second boolean indicates whether the learner goals
+        limit of the user has been exceeded.
+    """
+    learnt_topic_ids = get_all_learnt_topic_ids(user_id)
+    goals_limit_exceeded = False
+    belongs_to_learnt_list = False
+
+    if topic_id not in learnt_topic_ids:
+        goals_limit_exceeded = (
+            learner_goals_services.mark_topic_to_learn(user_id, topic_id))
+        belongs_to_learnt_list = False
+    else:
+        belongs_to_learnt_list = True
+
+    return (
+        belongs_to_learnt_list,
+        goals_limit_exceeded)
+
+
+def _remove_topic_ids_from_learner_goals(
```

Comment:
Why not use the remove function in learner_goals_services? If you need to handle multiple topic IDs, make that function accept a list and generalize it (with get_multi). Since, now there are 2 functions that call `learner_goals.remove_topic_id_from_learn` and both seem to be doing the same thing.

Reply:
Done.